### PR TITLE
Add `dq.eingenstates()` and `dq.isherm()`

### PR DESCRIPTION
**Pull request by RemiRousseau**
This pull request proposed to add eigenstates and isherm functions to utils of dynamiqs.

### CODE REVIEW

File: dynamiqs/utils/utils.py [hunk 1]
```diff
@@ -729,6 +752,45 @@ def fidelity(x: ArrayLike, y: ArrayLike) -> Array:
         return _dm_fidelity_gpu(x, y)
 
 
+def eigenstates(
+    x: ArrayLike, sort: Literal['high', 'low'] = 'low'
```
[the remark this answers is not in the document]
Comment (RemiRousseau) [reply dynamiqs/utils/utils.py]: Thanks for the inputs ! It should be fixed now

File: dynamiqs/utils/utils.py [hunk 2]
```diff
@@ -729,6 +752,45 @@ def fidelity(x: ArrayLike, y: ArrayLike) -> Array:
...
+        >>> D
+        Array([ 0.,  0.,  2.,  6., 12.], dtype=float32)
+    """
+    x = jnp.asarray(x)
+    if x.shape[-1] != x.shape[-2]:
+        raise ValueError('Argument `x` must be a square matrix.')
+
+    if isherm(x):
+        P, D = jax.lax.linalg.eigh(x, sort_eigenvalues=True)
+        if sort == 'high':
```
[the remark this answers is not in the document]
Comment (RemiRousseau) [reply dynamiqs/utils/utils.py]: Nice catch, indeed jax.lax.cond does the trick.

**Review by gautierronan — APPROVED** [review]
Thanks a lot for contributing @RemiRousseau! I made a few final changes (made the docstrings a bit more descriptive, and also added tolerences to dq.isherm). Merging this now :)

If you want to follow up with `dq.eigenenergies` in a separate PR, that would be very welcome, but no pressure ofc.

Thanks again !

**Comment by pierreguilmin** [comment]
Congrats on your first contribution! 🥳

**Pull request merged**